src/notify.coffee

```diff
@@ -15,6 +15,9 @@
 
 module.exports = (robot) ->
   robot.router.post '/notify/:target', (req, res) ->
+    if req.body.token isnt process.env.NOTIFY_TOKEN or ''
```

Comment:
I think we should first check that `process.env.NOTIFY_TOKEN` exists for backward compatibility:
```coffee
if process.env.NOTIFY_TOKEN? and req.body.token isnt process.env.NOTIFY_TOKEN
```
So if `NOTIFY_TOKEN` is undefined, it accepts any request.